app/views/results/common/_marks_summary_table.html.erb

```diff
@@ -7,7 +7,7 @@
   </thead>
   <% @mark_criteria.each do |criterion| -%>
     <%
-      mark = marks_map[criterion.id]
+      mark = marks_map[[criterion.class.to_s, criterion.id]]
```

Comment:
This was a really cool idea Christine had which made the code changes significantly smaller. Since its a string/int pair, that makes key-lookup stable.